src/compressor/zlib/CompressionZlib.cc

```diff
@@ -134,7 +134,7 @@ int CompressionZlib::decompress(const bufferlist &in, bufferlist &out)
       strm.avail_out = max_len;
       strm.next_out = c_out;
       ret = inflate(&strm, Z_NO_FLUSH);
-      if (ret != Z_OK && ret != Z_STREAM_END) {
```

Comment:
@yuyuyu101 Condition is
while (strm.avail_out == 0)
And it become false, if we get Z_BUF_ERROR (according zlib manual), so we exit from the loop.